test/upgrade/prober/prober.go

```diff
@@ -59,20 +59,21 @@ func RunEventProber(ctx context.Context, log *zap.SugaredLogger, client *testlib
 // AssertEventProber will send finish event and then verify if all events propagated well
 func AssertEventProber(ctx context.Context, t *testing.T, prober Prober) {
 	prober.Finish(ctx)
+	defer prober.remove()
 
 	waitAfterFinished(prober)
 
-	errors, events := prober.Verify(ctx)
-	if len(errors) == 0 {
-		t.Logf("All %d events propagated well", events)
+	eventErrs, eventCount, err := prober.Verify(ctx)
+	if err != nil {
+		t.Fatal("fetch error:", err)
```

Comment:
One nit. The error is already formulated. It would be better to just:

```golang
t.Fatal(err)
```

Reply:
Possibly. On the other hand, the `error` is more visible in the log. :
```
prober.go:69: fetch error: report fetched too early, receiver is in active state
```
vs.
```
prober.go:69: report fetched too early, receiver is in active state
```